jupyterhub/handlers/base.py

```diff
@@ -779,7 +782,14 @@ async def auth_to_user(self, authenticated, user=None):
         if not self.authenticator.enable_auth_state:
             # auth_state is not enabled. Force None.
             auth_state = None
+
+        if self.authenticator.manage_groups:
+            # Run authenticator user-group reload hook
+            user_groups = await self.load_user_groups(authenticated, user)
+            user.sync_groups(user_groups)
```

Comment:
As in other refresh fields, I think if the return value is `None`, it should mean no change, rather than empty the list:

```suggestion
            if user_groups is not None:
                user.sync_groups(user_groups)
```